modules/local/fetchtool_reads.nf

```diff
@@ -32,11 +32,11 @@ process FETCHTOOL_READS {
     library_strategy=\$(echo "\$(grep ${reads_accession} download_folder/${study_accession}/${study_accession}.txt | cut -f 7)" | tr '[:upper:]' '[:lower:]')
     library_layout=\$(echo "\$(grep ${reads_accession} download_folder/${study_accession}/${study_accession}.txt | cut -f 5)" | tr '[:upper:]' '[:lower:]')
 
-    export metadata_platform=\$(echo "\$(grep ${reads_accession} download_folder/${study_accession}/${study_accession}.txt | cut -f 8)" | tr '[:upper:]' '[:lower:]')
-    if [[ \$metadata_platform == "minion" || \$metadata_platform == "promethion" || \$metadata_platform == "gridion" ]]; then
+    export metadata_platform=\$(echo "\$(grep ${reads_accession} download_folder/${study_accession}/${study_accession}.txt | cut -f 9)" | tr '[:upper:]' '[:lower:]')
+    if [[ \$metadata_platform == "oxford_nanopore" ]]; then
         platform="ont"
-    elif [[ \$metadata_platform == "pacbio rs" || \$metadata_platform == "pacbio rs ii" ]]; then
-        platform="pacbio"
+    elif [[ \$metadata_platform == "pacbio_smrt" ]]; then
+        platform="pb"
```

Comment:
I might be missing something very stupid here, but why are these variables being exported? The platform isn't being used as an output for anything, or sent to an output file or anything. Same for the library_layout and library_strategy above it

Reply:
I'll answer to both questions about these odd variables here :)

Especially at the beginning we were only relying on the fetchtool to obtain sequence metadata. Now it's a bit blurry in my head, as we can also get private data with FIRE and I am not sure whether the fetchtool will be a thing in the long term. But the point is that you could originally pick EITHER the fetchtool OR the samplesheet. However, we do not have a priori knowledge of the sample, unless we manually go and check on ENA. The choice of the workflow / assembler / certain parameters depends on these metadata, which are appended to meta in this module (line 13). For long reads, especially the platform is important to identify what subworflow is best suited for your data, but the evaluation only happens in `long_read_assembler.nf`. The library variables are instead for short reads.

Does this answer to the question?